src/network/network_client.h

```diff
@@ -112,9 +112,15 @@ typedef ClientNetworkGameSocketHandler MyClient;
 void NetworkClient_Connected();
 void NetworkClientSetCompanyPassword(const char *password);
 
-extern CompanyID _network_join_as;
+/** Information required to join a server. */
+struct NetworkJoinInfo {
+	NetworkJoinInfo() : company(COMPANY_SPECTATOR), server_password(nullptr), company_password(nullptr) {}
+	NetworkAddress address;          ///< The address of the server to join.
+	CompanyID      company;          ///< The company to join.
+	const char    *server_password;  ///< The password of the server to join.
+	const char    *company_password; ///< The password of the company to join.
```

Comment:
Please don't align like this; it is a nightmare to do when types change .. personally aligning for comments is already "meh" (but we do it everywhere, so I guess we should), but for types is just even worse :P